api/src/opentrons/protocol_api/instrument_context.py

```diff
@@ -1578,3 +1582,93 @@ def __repr__(self) -> str:
 
     def __str__(self) -> str:
         return "{} on {} mount".format(self._core.get_display_name(), self.mount)
+
+    @requires_version(2, 15)
+    def configure_for_volume(self, volume: float) -> None:
+        """Configure the pipette to handle a specific volume of liquid.
+
+        Some pipettes have multiples of operation to handle certain volumes of liquid accurately.
+        For instance, the 50 µL GEN3 Single and Eight Channel pipettes must be placed in a low
+        volume mode to accurately dispense 1 µL of liquid. This function sets the mode of a pipette
+        - if applicable - appropriately for the specified volume.
+
+        The function can only be called when no liquid is aspirated into the pipette. It can be called
+        whether or not a tip is present, though when the pipette is in a certain mode some tips may be
+        unavailable.
+
+        The function must be called to put the pipette in the proper mode for a later aspiration. The
+        mode then persists until the next time the function is called.
+
+        Pipette modes also determine other properties of the pipette such as default flow rates and
+        maximum or minimum handlable volumes. The properties of this object may change when you change
+        modes.
+
+        .. note ::
+
+            Changing a pipette's mode will reset its flowrates.
+
+        If you are executing a protocol with many different volumes, it's a good practice to call this
+        function immediately before every distinct transfer or aspirate, using the volumes that you are
+        about to transfer or aspirate. This style also means the mode will always be correct when
+        executing from a volume list:
+
+        .. code-block:: python
+
+            tiprack = protocol.load_labware('opentrons_flex_96_tiprack_50ul', 'D1')
+            sample_plate = protocol.load_labware('nest_1_reservoir_195ml', 'D2')
+            instr = protocol.load_instrument('p50_single_gen3', 'left', tip_racks=[tiprack])
+            # this list might have been loaded from a CSV
+            for volume in [1, 2, 3, 4, 1, 5, 2, 8]:
+                instr.pick_up_tip()
+                # configuring for each volume before aspiration ensures the pipette is always in the
+                # right mode
+                instr.configure_for_volume(volume)
+                instr.aspirate(volume=volume, location=sample_plate['A1'])
+                instr.dispense(location=sample_plate['A1'])
+                instr.drop_tip()
```

Comment:
Nitpick: `tip_rack` instead of `tiprack`

```suggestion
            tip_rack = protocol.load_labware('opentrons_flex_96_tiprack_50ul', 'D1')
            sample_plate = protocol.load_labware('nest_1_reservoir_195ml', 'D2')
            instr = protocol.load_instrument('p50_single_gen3', 'left', tip_racks=[tip_rack])
            # this list might have been loaded from a CSV
            for volume in [1, 2, 3, 4, 1, 5, 2, 8]:
                instr.pick_up_tip()
                # configuring for each volume before aspiration ensures the pipette is always in the
                # right mode
                instr.configure_for_volume(volume)
                instr.aspirate(volume=volume, location=sample_plate['A1'])
                instr.dispense(location=sample_plate['A1'])
                instr.drop_tip()
```